File: src/Gastromio.Persistence.MongoDB/DishCategoryModel.cs
using System;
using System.Collections.Generic;

namespace Gastromio.Persistence.MongoDB
{
    public class DishCategoryModel
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public int OrderNo { get; set; }

        public bool? Enabled { get; set; }

        public IList<DishModel> Dishes { get; set; }
    }
}

File: src/Gastromio.App/ClientApp/src/app/shared/models/dish-category.model.ts
import {DishModel} from './dish.model';

export class DishCategoryModel {

  constructor(init?: Partial<DishCategoryModel>) {
    if (init) {
      Object.assign(this, init);
    }
    this.dishes = this.dishes?.map(dish => new DishModel(dish));
  }

  public id: string;

  public name: string;

  public enabled: boolean;

  public dishes: Array<DishModel>;

  public clone(): DishCategoryModel {
    return new DishCategoryModel({
      id: this.id,
      name: this.name,
      enabled: this.enabled,
      dishes: this.dishes?.map(dish => dish?.clone())
    });
  }

}
